### ISSUE TITLE
cfn-lint incorrectly reports the line/column number of found errors

### ISSUE THREAD

**Issue by dbaghdanov** [aws-cloudformation/cfn-lint#2287]
*cfn-lint version: cfn-lint 0.61.0*

*Description of issue.*
cfn-lint stopped correctly listing the LineNumber ColumnNumber between version 0.58.0 and 0.61.0

Please provide as much information as possible:
* Template linting issues:
```yaml
  SomeLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: myproject
      Runtime: dotnet6
      FunctionName: myLambdaFunction
      CodeUri: ./src/my-project/
      INVALID:    ### <<== this line should report an error
```

0.58.0 output:
```
E0000 Null value at line 70 column 15
template.yaml:70:15
```

0.61.0 output:
```
E0001 Error transforming template: Resource with id [SomeLambdaFunction] is invalid. property INVALID not defined for resource of type AWS::Serverless::Function
template.yaml:1:1
```

**Comment by PatMyron** [comment]
likely related to https://github.com/aws-cloudformation/cfn-lint/pull/2242, null values are allowed in some Serverless property types and not others, so the Serverless transform error is returned as-is now

**Comment by dbaghdanov** [comment]
I receive the same error even if I use a non-null value:

```yaml
SomeLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: myproject
      Runtime: dotnet6
      FunctionName: myLambdaFunction
      CodeUri: ./src/my-project/
      INVALID:    'this is an invalid property of serverless::function'
```

outputs:
```
E0001 Error transforming template: Resource with id [SomeLambdaFunction] is invalid. property INVALID not defined for resource of type AWS::Serverless::Function
.\template.yaml:1:1
```

But my point is that the error isn't at 1:1, its at x:y

**Comment by kddejong** [comment]
Need to dig into this more.  Because we don't have spec for SAM we have to transform the template to "valid" CloudFormation.  Since the template is transformed we some times lose line numbers and we are subject to what SAM transform package provides back to us.

**Comment by kddejong** [comment]
At this point I will call this won't fix.  Since we have to validate the transformed template we lose a lot of the original line numbers.  We do our best to keep as many of them as we can during transformation but the results are based on how SAM works. 

The other option may be to flip to adding in the schemas for the resource types and then we don't have to do a transform at all.  The problem with that is it adds some other complexity with ref/getatts to resources we don't see existing.
